### ISSUE TITLE
`document.opened`-event seems to be not handled (or not triggered)

### ISSUE THREAD

**Issue by mbaas2** [egodigital/vscode-powertools#20]
I'm trying to write some code that would be executed when a file was opened. So I took a slow start and tried your sample that responds to `file.changed`. Once I had that working, I simply went ahead and changed the `type`-param to `document.opened` - and nothing happens when I open a file of the specified type (which previously responded fine to `file.changed`.)

**Comment by ghost** [comment]
@mbaas2 

The wiki documentation was incomplete, but has now including a [code example](https://github.com/egodigital/vscode-powertools/wiki/Events#documentopened) for that event.

The `args` implements [DocumentOpenedEventActionScriptArguments](https://egodigital.github.io/vscode-powertools/api/interfaces/_contracts_.documentopenedeventactionscriptarguments.html) interface, which contains a new shorthand property called [file](https://egodigital.github.io/vscode-powertools/api/interfaces/_contracts_.documentopenedeventactionscriptarguments.html#file) since version 0.38.1.

Keep in mind to updated to [Visual Studio Code 1.33](https://code.visualstudio.com/updates/v1_33), which is minimum requirement for the extension.

**Comment by mbaas2** [comment]
Thanks. Got it working now! :-)
BUT there is one issue where even Google did not help: I have set `"files": [ "**/*.mipage"]` and it reported that `foo.mipage.git` was opened, too. It's not visible in the editor, but probably needed for the git-integration. But I was wondering if there's a way to modify the pattern to avoid it responding to that. However, I did not find any mention of "line end" or "end of string"-matching in glob-patterns, it seems to be implicit. So...is there a way to avoid this match?

**Comment by ghost** [comment]
@mbaas2 

Can you post some example settings and code?

**Comment by mbaas2** [comment]
Sure:

**settings.json:**
```
{
    "ego.power-tools": {
        "events": [
            {
                "type": "document.opened",
                "files": [
                    "**/*.mipage",
                    "**/*.dyalog"
                ],
                "exclude": [
                    ""
                ],
                "action": {
                    "script": "onDocumentOpened.js",
                    "type": "script"
                }
            }
        ]
    }
}

```
**onDocumentOpened.js:**
```
exports.execute = async(args) => {
    // args => https://egodigital.github.io/vscode-powertools/api/interfaces/_contracts_.documentopenedeventactionscriptarguments.html

    const path = require('path');

    // s. https://code.visualstudio.com/api/references/vscode-api
    const vscode = args.require('vscode');


    vscode.window.showInformationMessage(
        `The following file has been opened: ${ 
            path.relative(
                __dirname + '/..',
                args.file.fsPath
            )
         }`
    );
};

```

**Comment by ghost** [comment]
@mbaas2 

Where are these files stored? Inside `.git` or `.vscode` subfolder?

**Comment by mbaas2** [comment]
`onDocumentOpened.js` and `settings.jxon` are in a `.vscode`-subfolder of my repository.
The `foo.mipage.git` file that is shown in the msg actually does not exist within that repo!

**Comment by ghost** [comment]
@mbaas2 

**The bug should be fixed since version 0.39.0 now.**

**Comment by mbaas2** [comment]
Yeah, works like a charm. Thanks very much!
